web/source/text/domKbdInterface.ts

```diff
@@ -26,7 +26,11 @@ namespace com.keyman.text {
    */    
   KeyboardInterface.prototype.registerStub = function(Pstub): number {
     let keyman = com.keyman.singleton;
-    return keyman.keyboardManager._registerStub(Pstub);
+    keyman.keyboardManager._registerStub(Pstub).then(result => {
+      return result
+    });
+
+    return null;
```

Comment:
Yeah, those `return` statements won't work.

Looks like the change may break this (legacy) API function.

Reply:
Is there a test page for this?

Reply:
So this is another API change that would need documentation? The number returned previously was used for jsonp-style wrapping of keyboard script loads, was it not? Is that still needed?

Reply:
Are JSON-style keyboard loading scripts the same as the ones used in unit tests?

Are they the same ones no longer supported in Keyman 10+ (as documented in keymanapp/help.keyman.com#435)?

Reply:
That's not what that PR appears to document.  We no longer auto-create the loaders that would use the legacy approach, but that doesn't mean we ever stated dropping support the approach itself.

Not that I'm against doing so; we'd just need to note it as a breaking change.

Reply:
Should we change and document this API to now return a Promise<number>?

Reply:
I would prefer not to break this API. Given it returns a unique number, we should be able to generate that synchronously and return that value without problem.